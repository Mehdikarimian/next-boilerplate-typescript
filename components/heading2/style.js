import styled from 'styled-components';

export const Wrapper = styled.h2`
  color: green;
`