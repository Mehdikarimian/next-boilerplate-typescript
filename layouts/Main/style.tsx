import styled from 'styled-components';

export const Container = styled.div`
  padding: 0px 0px;
`