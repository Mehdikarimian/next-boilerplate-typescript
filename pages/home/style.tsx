import styled from 'styled-components';
export const Wrapper = styled.div`
h2 {
  font-size: 30px;
}
`