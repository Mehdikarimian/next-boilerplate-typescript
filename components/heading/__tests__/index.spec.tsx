import React from 'react'
import 'jest-styled-components'
import { shallow } from 'enzyme'
import Heading from '../index'
import { Wrapper } from '../style';
import renderer from 'react-test-renderer'
import 'jest-styled-components'

describe('With Enzyme', () => {
  it('Expect to not log errors in console', () => {
    const spy = jest.spyOn(global.console, 'error');
    shallow(<Heading text="Hello world!" />);
    expect(spy).not.toHaveBeenCalled();
  });

  it('Component shows "Hello world!"', () => {
    const component = shallow(<Heading text="Hello world!"></Heading>);
    expect(component.find('h2').text()).toEqual('Hello world!')
  })

  it('Should render and match the snapshot', () => {
    const app = shallow(<Heading text="Hello world!" />);
    expect(app).toMatchSnapshot();
  });
})

describe('Jest Styled Components', () => {
  const tree = renderer.create(<Wrapper />).toJSON()
  expect(tree).toMatchSnapshot()
  expect(tree).toHaveStyleRule('color', 'red')
})