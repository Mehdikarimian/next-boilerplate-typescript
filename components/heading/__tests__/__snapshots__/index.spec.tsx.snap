// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
.c0 {
  color: red;
}

<div
  className="c0"
/>
`;

exports[`With Enzyme Should render and match the snapshot 1`] = `ShallowWrapper {}`;
